cmake/sdp_asm_check.cmake

```diff
@@ -11,17 +11,17 @@ function(asm_check)
     get_filename_component(src_dir ${hrt_src} DIRECTORY)
 
     execute_process(COMMAND ${CMAKE_COMMAND} -E compare_files --ignore-eol
-                    ${src_dir}/${asm_filename}.s ${asm_filename}-temp.s
+                    ${src_dir}/${asm_filename}-${target_soc}.s ${asm_filename}-${target_soc}-temp.s
```

Comment:
`target_soc` is a required argument to the CMake script, please ensure proper validation.